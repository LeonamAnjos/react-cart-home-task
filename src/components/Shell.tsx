import React, { ReactNode } from "react";
import AppBar from "@mui/material/AppBar";
import Container from "@mui/material/Container";
import CssBaseline from "@mui/material/CssBaseline";
import Toolbar from "@mui/material/Toolbar";

type ShellProps = {
  toolbarContent: ReactNode;
  children: ReactNode;
};

const Shell = (props: ShellProps) => {
  const { toolbarContent, children } = props;

  return (
    <>
      <CssBaseline />

      <AppBar position="static">
        <Toolbar>{toolbarContent}</Toolbar>
      </AppBar>

      <Container>{children}</Container>
    </>
  );
};

export default Shell;
