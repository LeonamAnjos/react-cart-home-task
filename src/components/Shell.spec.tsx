import { render } from "@testing-library/react";
import Shell from "./Shell";

describe("ElementCard", () => {
  it("WHEN rendered", () => {
    const { container } = render(
      <Shell toolbarContent={"Toolbar content test"}>{"Content test"}</Shell>
    );

    expect(container).toMatchSnapshot();
  });
});
