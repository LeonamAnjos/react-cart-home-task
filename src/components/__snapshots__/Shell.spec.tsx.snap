// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementCard WHEN rendered 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      Toolbar content test
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    Content test
  </div>
</div>
`;
